EXOMessageClassification: Cannot create anymore

### Description of the issue

@FabienTschanz Hi, you changed this resource to silently continue if it cannot find the object by Identity, problem is that it tries afterwards to get it by DisplayName instead but you missed changing that one too.

https://github.com/microsoft/Microsoft365DSC/commit/a14c3b5140c0e892f9526792ddf5b46c0dcb68a7#diff-8015e719a5e748f7de739773b8c8b038dde2f232ea1cdc5509fbeb180451fc58

```powershell
            $MessageClassification = Get-MessageClassification -Identity $Identity -ErrorAction SilentlyContinue

            if ($null -eq $MessageClassification)
            {
                if (-not [System.String]::IsNullOrEmpty($DisplayName))
                {
                    Write-Verbose -Message "Couldn't retrieve Message Classification policy by Id {$($Identity)}. Trying by DisplayName."
                    $MessageClassification = Get-MessageClassification -Identity $DisplayName # <--- -ErrorAction SilentlyContinue is missing here
                }
                if ($null -eq $MessageClassification)
                {
                    return $nullReturn
                }
            }
        }
        else
        {
            $MessageClassification = $Script:exportedInstance
        }
```

### Microsoft 365 DSC Version

1.25.1112.1

### Which workloads are affected

Exchange Online

### The DSC configuration

```powershell
EXOMessageClassification "EXOMessageClassification-EXOMessageClassification_1"
        {
            ApplicationId               = $EXOApplicationId;
            CertificateThumbprint       = $EXOCertThumbprint;
            DisplayName                 = "EXOMessageClassification_1";
            DisplayPrecedence           = "Highest";
            Ensure                      = "Present";
            Identity                    = "EXOMessageClassification_1";
            Name                        = "EXOMessageClassification_1";
            PermissionMenuVisible       = $True;
            RecipientDescription        = "Shown to recipients";
            RetainClassificationEnabled = $True;
            SenderDescription           = "Shown to senders";
            TenantId                    = $OrganizationName;
        }
```

### Verbose logs showing the problem

```powershell
[[EXOMessageClassification]EXOMessageClassification-EXOMessageClassification_1]
VERBOSE: [8cacac82e000000]: LCM:  [ Start  Test     ]  
[[EXOMessageClassification]EXOMessageClassification-EXOMessageClassification_1]
VERBOSE: [8cacac82e000000]:                            
[[EXOMessageClassification]EXOMessageClassification-EXOMessageClassification_1] Testing configuration of the 
EXOMessageClassification with Identity {EXOMessageClassification_1}
VERBOSE: [8cacac82e000000]:                            
[[EXOMessageClassification]EXOMessageClassification-EXOMessageClassification_1] Getting Message Classification 
Configuration for EXOMessageClassification_1
VERBOSE: [8cacac82e000000]:                            
[[EXOMessageClassification]EXOMessageClassification-EXOMessageClassification_1] Couldn't retrieve Message 
Classification policy by Id {EXOMessageClassification_1}. Trying by DisplayName.
[2025-11-15 13:35:45] -  
##[error]Deploy execution failed ❌: ||The operation couldn't be performed because object 'EXOMessageClassification_1' couldn't be found on 'PR3P195A06DC003.EURP195A006.PROD.OUTLOOK.COM'.
##[error]||The operation couldn't be performed because object 'EXOMessageClassification_1' couldn't be found on 'PR3P195A06DC003.EURP195A006.PROD.OUTLOOK.COM'.
##[error]PowerShell exited with code '1'.
```

### Environment Information + PowerShell Version

```powershell
Win11/PS5.1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,13 @@
 # Change log for Microsoft365DSC
 
+# UNRELEASED
+
+* EXOMessageClassification
+  * Fixed issue where the resource would fail during export if it could not
+    find it by `DisplayName`, and it needed to be created, by silently
+    continuing if that is the case
+    FIXES [#6691](https://github.com/microsoft/Microsoft365DSC/issues/6691)
+
 # Release 1.25.1112.1
 
 * AADActivityBasedTimeoutPolicy

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_EXOMessageClassification/MSFT_EXOMessageClassification.psm1
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_EXOMessageClassification/MSFT_EXOMessageClassification.psm1
@@ -112,7 +112,7 @@ function Get-TargetResource
                 if (-not [System.String]::IsNullOrEmpty($DisplayName))
                 {
                     Write-Verbose -Message "Couldn't retrieve Message Classification policy by Id {$($Identity)}. Trying by DisplayName."
-                    $MessageClassification = Get-MessageClassification -Identity $DisplayName
+                    $MessageClassification = Get-MessageClassification -Identity $DisplayName -ErrorAction SilentlyContinue
                 }
                 if ($null -eq $MessageClassification)
                 {

